sources/Interop/D3D12MemoryAllocator/D3D12MemAlloc.Init.cs

```diff
@@ -0,0 +1,73 @@
+        ///     All assertions are disabled.
+        /// #define D3D12MA_DEBUG_LEVEL 1
+        ///     Enable basic assertions.
+        /// #define D3D12MA_DEBUG_LEVEL 2
+        ///     Enable all assertions, including those in hot paths. This guarantees maximum assertion
+        ///     coverage, but it might have a noticeable impact on performance when using the library.
+        /// </code>
+        /// </summary>
+        public static readonly uint D3D12MA_DEBUG_LEVEL = get_app_context_data(nameof(D3D12MA_DEBUG_LEVEL), IsDebug ? 1 : 0);
+
+        /// <summary>
+        /// When defined to value other than 0, the library will try to use
+        /// <see cref="D3D12_SMALL_RESOURCE_PLACEMENT_ALIGNMENT"/> or <see cref="D3D12_SMALL_MSAA_RESOURCE_PLACEMENT_ALIGNMENT"/>
+        /// for created textures when possible, which can save memory because some small textures
+        /// may get their alignment 4K and their size a multiply of 4K instead of 64K.
+        /// <code>
+        /// #define D3D12MA_USE_SMALL_RESOURCE_PLACEMENT_ALIGNMENT 0
+        ///     Disables small texture alignment.
+        /// #define D3D12MA_USE_SMALL_RESOURCE_PLACEMENT_ALIGNMENT 1
+        ///     Enables conservative algorithm that will use small alignment only for some textures
+        ///     that are surely known to support it.
+        /// #define D3D12MA_USE_SMALL_RESOURCE_PLACEMENT_ALIGNMENT 2
+        ///     Enables query for small alignment to D3D12 (based on Microsoft sample) which will
+        ///     enable small alignment for more textures, but will also generate D3D Debug Layer
+        ///     error #721 on call to ID3D12Device::GetResourceAllocationInfo, which you should just
+        ///     ignore.
+        /// </code>
+        /// </summary>
+        public static readonly uint D3D12MA_USE_SMALL_RESOURCE_PLACEMENT_ALIGNMENT = get_app_context_data(nameof(D3D12MA_USE_SMALL_RESOURCE_PLACEMENT_ALIGNMENT), 1);
+
+        /// <summary>Minimum alignment of all allocations, in bytes. Set to more than 1 for debugging purposes only. Must be power of two.</summary>
+        public static readonly ulong D3D12MA_DEBUG_ALIGNMENT = get_app_context_data(nameof(D3D12MA_DEBUG_ALIGNMENT), 1);
+
+        /// <summary>Minimum margin before and after every allocation, in bytes. Set nonzero for debugging purposes only.</summary>
+        public static readonly ulong D3D12MA_DEBUG_MARGIN = get_app_context_data(nameof(D3D12MA_DEBUG_MARGIN), 0);
+
+        /// <summary>Define this macro to 0 to disable usage of DXGI 1.4 (needed for <see cref="IDXGIAdapter3"/> and query for memory budget).</summary>
+        public static readonly uint D3D12MA_DXGI_1_4 = get_app_context_data(nameof(D3D12MA_DXGI_1_4), 1);
+
+        /// <summary>Default size of a block allocated as single <see cref="ID3D12Heap"/>.</summary>
+        public static readonly ulong D3D12MA_DEFAULT_BLOCK_SIZE = get_app_context_data(nameof(D3D12MA_DEFAULT_BLOCK_SIZE), 256U * 1024 * 1024);
```

Comment:
The type  needs to be `UL` so the app context is parsed as `ulong` and not `uint` plus implicit upcast
```suggestion
        public static readonly ulong D3D12MA_DEFAULT_BLOCK_SIZE = get_app_context_data(nameof(D3D12MA_DEFAULT_BLOCK_SIZE), 256UL * 1024 * 1024);
```

Reply:
Fixed in 7d56a15348d821fee7be2003c0f30ed12fbcc15c for all `ulong` settings 👍